File: 14-longest-common-prefix/longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        
        Arrays.sort(strs);

        String first = strs[0];
        String last = strs[strs.length-1];

        int min = first.length() < last.length() ? first.length() : last.length();
        int j = 0;
        for(int i=0; i< min ;i++)
        {
            if(first.charAt(i) != last.charAt(i))
            break;
            j++;
        }

        return strs[0].substring(0,j);
    }
}

File: 14-longest-common-prefix/longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        sorted_strs = sorted(strs)
        first = sorted_strs[0]
        last = sorted_strs[len(strs)-1]
        a, b = len(first), len(last)
        if a < b:
            mini = a
        else:
            mini = b
        k = mini
        for i in range(mini):
            if first[i] != last[i]:
                k = i
                break
        
        return strs[0][:k]   
